crates/amaru/src/stages/mod.rs

```diff
@@ -50,16 +52,37 @@ pub mod pull;
 
 pub type BlockHash = pallas_crypto::hash::Hash<32>;
 
+/// Whether or not data is stored on disk or in memory.
+#[derive(Clone, Debug)]
+pub enum StorePath {
+    InMem,
+    OnDisk(PathBuf),
+}
+
 pub struct Config {
-    pub ledger_dir: PathBuf,
-    pub chain_dir: PathBuf,
+    pub ledger_store: StorePath,
```

Comment:
I'm wondering if this choice has to be exposed at the configuration level. In memory stores are probably something only relevant for testing scenarios where we manually wore things?

Reply:
`Config` is the result of parsing `Args` and not exposed to the user: It can only be set programmatically. While I agree this is mostly useful for testing, it seems to me we shall be using this extensively in the near future and there might be use cases for in-memory only nodes, so I would like to keep this option open.